### PR TITLE
[setup-kubernetes.sh] Add option for specifying credentials to docker.io for pulling images

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Bugfix

### Description

Currently our PR jobs and nightlies are failing because of reaching the pull limit from docker hub, when we are deploying `minikube`.

This PR adding option to specify the `DOCKER_LOGIN` env, which value should be in the pattern `username:password` and encoded in Base64. By this, we will be able to log in and pull the image without problem with reaching the pull limit.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Review by scholzj — COMMENTED** [review]
Can you explain what it is and why is it needed?

**Comment by im-konge** [comment]
@scholzj already done (hopefully)

**Comment by scholzj** [comment]
So, how is this image maintained? Is there some documentation for it? Also, do you use other things to limit the pull limit issue? Such as login into Docker Hub?

**Comment by im-konge** [comment]
Even when I logged in, I reached the limit. The main org is here https://quay.io/repository/openshift/community-e2e-images .I didn't find any documentation yet, but it seems that it's active (last push was 7 days ago, upstream docker registry image is 14 days old).

Also I was thinking to push the image to some other our org/repo, but it's not necessary from MPOV.

**Review by Frawless — APPROVED** [review]

**Review by scholzj — APPROVED** [review]

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

File: .azure/scripts/setup-kubernetes.sh [hunk 1]
```diff
@@ -80,6 +80,25 @@ if [ "$TEST_CLUSTER" = "minikube" ]; then
...
+
+    if [ -n "$DOCKER_LOGIN" ]
+    then
+      docker exec "minikube" bash -c "echo '{
+        \"auths\": {
+          \"https://index.docker.io/v1/\": {
+            \"auth\": \"'$DOCKER_LOGIN'\"
+          }
+        }
+      }' | sudo tee -a /var/lib/kubelet/config.json > /dev/null && sudo systemctl restart kubelet"
```
Comment (Frawless) [inline .azure/scripts/setup-kubernetes.sh:97]: Wouldn't be better to copy the whole auth configuration for docker from VM?

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectBuilderST

**Comment by im-konge** [comment]
@Frawless @scholzj copying of the docker config is now correct, minikube is able to pull the registry image without any problem. Can you do review again, please?

File: .azure/scripts/setup-kubernetes.sh [hunk 2]
```diff
@@ -80,6 +81,18 @@ if [ "$TEST_CLUSTER" = "minikube" ]; then
     fi
 
     minikube addons enable default-storageclass
+
+    # Add Docker hub credentials to Minikube
+
+    set +ex
```
Comment (scholzj) [inline .azure/scripts/setup-kubernetes.sh:87]: Should the `set +ex` / `set -ex` be inside the `if`?

Comment (im-konge) [reply .azure/scripts/setup-kubernetes.sh:87]: Yeah, right

File: .azure/scripts/setup-kubernetes.sh [hunk 3]
```diff
@@ -80,6 +81,17 @@ if [ "$TEST_CLUSTER" = "minikube" ]; then
...
+    if [ "$COPY_DOCKER_LOGIN" = "true" ]
+    then
+      set +ex
+
+      docker exec "minikube" bash -c "echo '$(cat $HOME/.docker/config.json)'| sudo tee -a /var/lib/kubelet/config.json > /dev/null && sudo systemctl restart kubelet"
+
+      set -ex
+    fi
+
 	  minikube addons enable registry
```
Comment (scholzj) [inline .azure/scripts/setup-kubernetes.sh:95]: Looks like something you did not break. But could you fix the alignment here?

**Review by scholzj — APPROVED** [review]

File: .azure/scripts/setup-kubernetes.sh (hunk 3 above)
Comment (im-konge) [reply .azure/scripts/setup-kubernetes.sh]: It seems that this is problem on more lines, let me fix it.

Comment (scholzj) [reply .azure/scripts/setup-kubernetes.sh]: Thanks

**Review by Frawless — APPROVED** [review]

**Pull request merged**